### DISCUSSION TITLE
Arduino Portenta - problem to define digital output pin

### DISCUSSION THREAD

**Discussion by mironkvk** [micropython/micropython#16319] [Using MicroPython] [answered]
I use micropython via OpenMV on Arduino Portenta. I cannot define pin for external diode (built in diodes works ok). The code is as follows:
import time
from machine import Pin
led = Pin(6, Pin.OUT)
while (True):
led.on()
time.sleep_ms(500)
led.off()
time.sleep_ms(500)
There is a notification: Type error: can't convert 'int' object to str implicitly.
Can anyone help me?
Miron

**Comment by peterhinch** [comment, marked as answer]
Please format your code as [described here](https://github.com/orgs/micropython/discussions/9111).

If you're using OpenMV firmware you might be best targetting the query at them. My guess at the source of the error is that their pin numbering convention is to use a string rather than an integer (when instantiating the `Pin`).

**Comment by mironkvk** [comment]
Thanks for fast response. It works now. Actually the problem was related to conversion between int and str. In my case the notation  str('PH7') solved the problem.
